src/stdio.jl

```diff
@@ -52,6 +77,36 @@ function watch_stream(rd::IO, name::AbstractString)
     end
 end
 
+function send_stdio(name)
+    if verbose::Bool && !haskey(task_local_storage(), :IJulia_task)
+        task_local_storage(:IJulia_task, "send $name task")
+    end
+    send_stream(name)
+end
+
+send_stdout(t::Timer) = send_stdio("stdout")
+send_stderr(t::Timer) = send_stdio("stderr")
+
+function send_stream(name::AbstractString)
+    buf = bufs[name]
+    if buf.size > 0
+        s = takebuf_string(buf)
+        #if the last char is not valid, maybe only the start of a partially encoded UTF character
+        #was written, so we write it back to the buffer in the hope that the rest of the character will arrive soon
+        last_ind = chr2ind(s, length(s))
+        last_chrstr = s[last_ind:sizeof(s)]
+        if !isvalid(last_chrstr)
+            write(buf, last_chrstr)
+            last_ind -= 1
+        end
+        if last_ind > 0
```

Comment:
Oh I see, you want people do be able to display non-string data - was that possible previously? In any case that's a good idea, and your functions look sexy. Do you want me to just add them and see how they go?